File: src/BruteForce/LinearSearch/Java/src/test/com/example/LinearSearchTest.java
package com.example;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

import java.util.Arrays;
import java.util.List;

public class LinearSearchTest {

    @Test
    void testFound() {
        assertEquals(2, LinearSearch.find(Arrays.asList(10, 23, 45, 70, 11, 15), 45));
        assertEquals(0, LinearSearch.find(Arrays.asList(1, 2, 3, 4, 5), 1));
        assertEquals(4, LinearSearch.find(Arrays.asList(1, 2, 3, 4, 5), 5));
    }

    @Test
    void testNotFound() {
        assertEquals(-1, LinearSearch.find(Arrays.asList(10, 23, 45, 70, 11, 15), 99));
        assertEquals(-1, LinearSearch.find(Arrays.asList(), 10));
    }

    @Test
    void testDuplicates() {
        assertEquals(2, LinearSearch.find(Arrays.asList(1, 2, 3, 3, 4), 3));
    }

    @Test
    void testSingleElement() {
        assertEquals(0, LinearSearch.find(Arrays.asList(5), 5));
        assertEquals(-1, LinearSearch.find(Arrays.asList(5), 1));
    }
}

File: src/BruteForce/LinearSearch/Go/LSearch/linear_search_test.go
// File: linear_search_test.go

package linearsearch_test

import (
	linearsearch "linearsearch/LSearch"
	"testing"
)

func TestFound(t *testing.T) {
	tests := []struct {
		list   []int
		target int
		want   int
	}{
		{[]int{10, 23, 45, 70, 11, 15}, 45, 2},
		{[]int{1, 2, 3, 4, 5}, 1, 0},
		{[]int{1, 2, 3, 4, 5}, 5, 4},
	}
	for _, tt := range tests {
		t.Run("", func(t *testing.T) {
			got := linearsearch.Find(tt.list, tt.target)
			if got != tt.want {
				t.Errorf("Find(%v, %v) want %v, got %v", tt.list, tt.target, tt.want, got)
			}
		})
	}
}

func TestNotFound(t *testing.T) {
	tests := []struct {
		list   []int
		target int
		want   int
	}{
		{[]int{10, 23, 45, 70, 11, 15}, 99, -1},
		{[]int{}, 10, -1},
	}
	for _, tt := range tests {
		t.Run("", func(t *testing.T) {
			got := linearsearch.Find(tt.list, tt.target)
			if got != tt.want {
				t.Errorf("Find(%v, %v) want %v, got %v", tt.list, tt.target, tt.want, got)
			}
		})
	}
}

func TestDuplicates(t *testing.T) {
	got := linearsearch.Find([]int{1, 2, 3, 3, 4}, 3)
	want := 2
	if got != want {
		t.Errorf("Find(%v, %v) want %v, got %v", []int{1, 2, 3, 3, 4}, 3, want, got)
	}
}

func TestSingleElement(t *testing.T) {
	tests := []struct {
		list   []int
		target int
		want   int
	}{
		{[]int{5}, 5, 0},
		{[]int{5}, 1, -1},
	}
	for _, tt := range tests {
		t.Run("", func(t *testing.T) {
			got := linearsearch.Find(tt.list, tt.target)
			if got != tt.want {
				t.Errorf("Find(%v, %v) want %v, got %v", tt.list, tt.target, tt.want, got)
			}
		})
	}
}
